app/projectsmodel.cpp

```diff
@@ -549,21 +564,20 @@ void ProjectsModel::onProjectDetachedFromMergin( const QString &projectFullName
   project.mergin = MerginProject();
   project.local.projectNamespace = QLatin1String();
 
-  QModelIndex editIndex = index( ix );
+  const QModelIndex editIndex = index( ix );
   emit dataChanged( editIndex, editIndex );
 
   // This project should also be removed from project list for remote project model types,
   // however, currently one needs to click on "My projects/Shared/Explore" and that sends
   // another listProjects request. In new list this project will not be shown.
-  // However, this option is not allowed in GUI anyways.
+  // However, this option is not allowed in GUI anyway.
 
 }
 
-void ProjectsModel::onProjectAttachedToMergin( const QString & )
+void ProjectsModel::onProjectAttachedToMergin( const QString &projectId )
 {
-  // To ensure project will be in sync with server, send listProjectByName request.
-  // In theory we could send that request only for this one project.
-  listProjectsByName();
+  // To ensure project will be in sync with server, send fetchProjectsByProjectId request.
+  fetchProjectsByProjectId( { projectId } );
```

Comment:
If you fetch state only of this one project, won't other local projects states get unknown (not calculated)? When requesting this API (in `fetchProjectsByProjectId`) we clear the projects model and so the resulting state of the other projects is now _likely_ unknown (I did not test this, just assumption from the code).